tests/Config/GlobalQuery.test.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { GlobalQuery } from '../../src/Config/GlobalQuery';
+
+describe('GlobalQuery tests', () => {
+    it('getInstance() should return the same object', () => {
+        const globalQuery1 = GlobalQuery.getInstance();
+        const globalQuery2 = GlobalQuery.getInstance();
+
+        expect(Object.is(globalQuery1, globalQuery2)).toEqual(true);
+    });
+
+    it('should have an empty source by default', () => {
+        expect(new GlobalQuery().isEmpty()).toEqual(true);
+    });
+
+    it.each(['', ' ', '\n', '\n     \n    ', '  \n    \n'])(
+        'should have empty source if line breaks and spaces were set in the query',
```

Comment:
I think 'should have empty source' is a bit ambiguous.
Perhaps 'should report that the source is empty'...

Reply:
Rephrased after `isEmpty()` -> `hasInstructions()` rename.